### PR TITLE
sidecar defaultEndPoints should be IPv6 compatible

### PR DESCRIPTION
**Pull request by kfaseela**
should be merged along with https://github.com/istio/istio/pull/40265
Issue: https://github.com/istio/istio/issues/40245

Signed-off-by: Faseela K <faseela.k@est.tech>

### CODE REVIEW

**Comment by kfaseela** [comment]
gencheck failure is unrelated to this PR and will be fixed by https://github.com/istio/api/pull/2446

**Review by hzxuzhonghu — APPROVED** [review]
We should support ipv6 CI in order to support ipv6， currently could only resolve issue one by one. Looks ipv6 is fully supported. Not sure any user use ipv6 in production.

**Comment by howardjohn** [comment]
We do have an ipv6 job in CI - folks contributing to ipv6 should focus on improving its test coverage.

**Review by linsun — APPROVED** [review]
+1 on others commenting on ipv6 CI.

**Comment by linsun** [comment]
@kfaseela can you provide a plan on supporting ipv6 CI? I didn't want it to block your PR but would be needed before this is released in an official istio release.

File: networking/v1alpha3/sidecar.proto [hunk 1]
```diff
@@ -561,8 +561,9 @@ message IstioIngressListener {
   // traffic should be forwarded to. This configuration can be used to
   // redirect traffic arriving at the bind `IP:Port` on the sidecar to a `localhost:port`
   // or Unix domain socket where the application workload instance is listening for
-  // connections. Arbitrary IPs are not supported. Format should be one of `127.0.0.1:PORT`, `0.0.0.0:PORT`
-  // (which will forward to the instance IP), or `unix:///path/to/socket`
+  // connections. Arbitrary IPs are not supported. Format should be one of `127.0.0.1:PORT`,
+  // `0.0.0.0:PORT`, `[::1]:PORT`, `[::]:PORT` (which will forward to the instance IP),
```
Comment (howardjohn) [inline networking/v1alpha3/sidecar.proto:565]: The `(which will forward to the instance IP),` part is now confusing

I would have a list

```
- Forward to isntance ip: 0.0.0.0:PORT, [::]:PORT
- Forward to localhost: ...
- Forward to UDS socket: unix:///blah
```

Comment (kfaseela) [reply networking/v1alpha3/sidecar.proto:565]: > The `(which will forward to the instance IP),` part is now confusing
> 
> I would have a list
> 
> ```
> - Forward to isntance ip: 0.0.0.0:PORT, [::]:PORT
> - Forward to localhost: ...
> - Forward to UDS socket: unix:///blah
> ```

Will fix this soon

**Comment by kfaseela** [comment]
> @kfaseela can you provide a plan on supporting ipv6 CI? I didn't want it to block your PR but would be needed before this is released in an official istio release.

I did not really think about it. 🙂 We were not using sidecar api so far, but otherwise ipv6 works fine for us. We just started trying out a new experimental feature that recently came in which needed sidecar api, and realized that there are certain things which required change w.r.t ipv6. Let me see if I can spare some time in upstreaming some of the ipv6 test suites.

**Pull request merged**